### PR TITLE
fix: trim before and after space in query editor and vrl function

### PR DESCRIPTION
**Pull request by bjp232004**
## Release Notes

- **New Features**
	- Enhanced SQL query handling across multiple components by trimming whitespace from user inputs, ensuring cleaner and more reliable query execution.
	- Added a `refreshTimezone` method to update grid columns and regenerate histogram data in the search functionality.

- **Bug Fixes**
	- Improved error handling and user feedback for session data retrieval and search requests.

These updates enhance the overall robustness and user experience within the application, particularly in handling search queries and alerts.

### CODE REVIEW

File: web/src/views/RUM/AppSessions.vue [hunk 1]
```diff
@@ -426,7 +426,7 @@ const getSessions = () => {
     store.state.zoConfig.timestamp_column
   }) as zo_sql_timestamp, min(start) as start_time, max(end) as end_time, min(user_agent_user_agent_family) as browser, min(user_agent_os_family) as os, min(ip) as ip, min(source) as source, session_id from "_sessionreplay" ${
     sessionState.data.editorValue.length
-      ? " where " + sessionState.data.editorValue
+      ? " where " + sessionState.data.editorValue.trimStart().trimEnd()
```
[the remark this answers is not in the document]
Comment (bjp232004) [reply web/src/views/RUM/AppSessions.vue:429]: not required

File: web/src/plugins/traces/Index.vue [hunk 2]
```diff
@@ -516,7 +516,7 @@ export default defineComponent({
 
     function buildSearch() {
       try {
-        let query = searchObj.data.editorValue;
+        let query = searchObj.data.editorValue.trimStart().trimEnd();
```
[the remark this answers is not in the document]
Comment (bjp232004) [reply web/src/plugins/traces/Index.vue]: added single function.

File: web/src/composables/useLogs.ts [hunk 3]
```diff
@@ -553,7 +555,10 @@ const useLogs = () => {
 
   const updateUrlQueryParams = () => {
     const query = generateURLQuery(false);
-    if(query.hasOwnProperty("type") && query.type == "search_history_re_apply"){
+    if (
+      query.hasOwnProperty("type") &&
+      query.type == "search_history_re_apply"
+    ) {
```
[the remark this answers is not in the document]
Comment (bjp232004) [reply web/src/composables/useLogs.ts]: this was done as a part of another pr

File: web/src/composables/useLogs.ts [hunk 4]
```diff
@@ -3826,7 +3834,10 @@ const useLogs = () => {
     }
 
     searchObj.shouldIgnoreWatcher = false;
-    if(queryParams.hasOwnProperty("type") &&  queryParams.type == "search_history_re_apply"){
+    if (
+      queryParams.hasOwnProperty("type") &&
+      queryParams.type == "search_history_re_apply"
+    ) {
```
[the remark this answers is not in the document]
Comment (bjp232004) [reply web/src/composables/useLogs.ts]: This was done.

File: web/src/composables/useLogs.ts [hunk 5]
```diff
@@ -3706,9 +3711,12 @@ const useLogs = () => {
...
       searchObj.data.queryResults.aggs = null;
-      if(router.currentRoute.value.query.hasOwnProperty("type") &&  router.currentRoute.value.query.type == "search_history_re_apply"){
-       delete router.currentRoute.value.query.type;
-        }   
+      if (
+        router.currentRoute.value.query.hasOwnProperty("type") &&
+        router.currentRoute.value.query.type == "search_history_re_apply"
+      ) {
+        delete router.currentRoute.value.query.type;
+      }
```
[the remark this answers is not in the document]
Comment (bjp232004) [reply web/src/composables/useLogs.ts]: Do not want to keep the type so deleting this property from the object.

**Review by omkarK06 — APPROVED** [review]

**Pull request merged**